File: backend/src/main/java/com/backend/petplace/domain/user/repository/UserRepository.java
package com.backend.petplace.domain.user.repository;

import com.backend.petplace.domain.user.entity.User;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {

  boolean existsByNickName(String name);

  boolean existsByEmail(String email);

  Optional<User> findByNickName(String nickName);
}

File: backend/src/main/java/com/backend/petplace/domain/user/repository/UserRepository.kt
package com.backend.petplace.domain.user.repository

import com.backend.petplace.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Long> {
    fun existsByNickName(name: String): Boolean

    fun existsByEmail(email: String): Boolean

    fun findByNickName(nickName: String): User?
}